x-pack/plugins/index_lifecycle_management/public/url_generator.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup } from 'kibana/public';
+import { UrlGeneratorsDefinition } from '../../../../src/plugins/share/public/';
+import {
+  getPoliciesListPath,
+  getPolicyCreatePath,
+  getPolicyEditPath,
+} from './application/services/navigation';
+import { MANAGEMENT_APP_ID } from '../../../../src/plugins/management/public';
+import { SetupDependencies } from './types';
+import { PLUGIN } from '../common/constants';
+
+export const ILM_URL_GENERATOR_ID = 'ILM_URL_GENERATOR_ID';
+
+export enum ILM_PAGES {
+  LIST = 'policies_list',
+  EDIT = 'policy_edit',
+  CREATE = 'policy_create',
+}
+
+export interface IlmPoliciesListUrlGeneratorState {
+  page: ILM_PAGES.LIST;
+  absolute?: boolean;
+}
+
+export interface IlmPolicyEditUrlGeneratorState {
+  page: ILM_PAGES.EDIT;
+  policyName: string;
+  absolute?: boolean;
+}
+
+export interface IlmPolicyCreateUrlGeneratorState {
+  page: ILM_PAGES.CREATE;
+  absolute?: boolean;
+}
+
+export type IlmUrlGeneratorState =
+  | IlmPoliciesListUrlGeneratorState
+  | IlmPolicyEditUrlGeneratorState
+  | IlmPolicyCreateUrlGeneratorState;
```

Comment:
I think we can compress these types in the following way:

```ts
export interface IlmUrlGeneratorState {
  page: 'policies_list' | 'policy_edit' | 'policy_create';
  policyName?: string;
  absolute?: boolean;
}

// then the switch statement further down becomes - with auto complete for string literals

switch (state.page) {
  case 'policy_create': {
    return `${await getAppBasePath(!!state.absolute)}${getPolicyCreatePath()}`;
  }
  case 'policy_edit': {
    return `${await getAppBasePath(!!state.absolute)}${getPolicyEditPath(
      state.policyName!
    )}`;
  }
  case 'policies_list': {
    return `${await getAppBasePath(!!state.absolute)}${getPoliciesListPath()}`;
  }
}

```

I think this makes the code a bit easier to understand - just because there is less of it :). What do you think?